libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2351,6 +2351,50 @@ string YulUtilFunctions::copyArrayFromStorageToMemoryFunction(ArrayType const& _
 	});
 }
 
+string YulUtilFunctions::bytesConcatFunction(vector<Type const*> const& _argumentTypes)
+{
+	string functionName = "bytes_concat";
+	size_t totalParams = 0;
+	vector<Type const*> targetTypes;
+	for (Type const* argumentType: _argumentTypes)
+	{
+		solAssert(
+			argumentType->isImplicitlyConvertibleTo(*TypeProvider::bytesMemory()) ||
+			argumentType->isImplicitlyConvertibleTo(*TypeProvider::fixedBytes(32)),
+			""
+		);
+		targetTypes.emplace_back(
+			argumentType->category() == Type::Category::FixedBytes ? argumentType : TypeProvider::bytesMemory()
```

Comment:
Is this correct for short string literals?

Reply:
`StringLiterals` are implicitly convertible to both `FixedBytes`(if short enough) and `bytes memory`. But I can change behaviour so that `targetType` for literal is fixed bytes with appropriate length.

Reply:
I think FixedBytes should be preferred if it works.

Reply:
Works, changed to `FixedBytes` in case of short literals. Also, tweaked the tests to have both long and short literal examples